diff --git a/otherlibs/unix/socketaddr.c b/otherlibs/unix/socketaddr.c
index b340eeb..a0d4577 100644
--- a/otherlibs/unix/socketaddr.c
+++ b/otherlibs/unix/socketaddr.c
@@ -121,7 +121,7 @@ value caml_unix_alloc_sockaddr(union sock_addr_union * adr /*in*/,
     CAMLreturn(alloc_unix_sockaddr(caml_alloc_string(0)));
   }
 
-  switch(adr->s_gen.sa_family) {
+  switch((int)adr->s_gen.sa_family) {
   case AF_UNIX:
     { /* Based on recommendation in section BUGS of Linux unix(7). See
          http://man7.org/linux/man-pages/man7/unix.7.html. */
