diff --git a/runtime/io.c b/runtime/io.c
index 5401af2d3a..0879a808f0 100644
--- a/runtime/io.c
+++ b/runtime/io.c
@@ -179,6 +179,10 @@ CAMLexport struct channel * caml_open_descriptor_in(int fd)
   channel->fd = fd;
   caml_enter_blocking_section_no_pending();
   channel->offset = lseek(fd, 0, SEEK_CUR);
+  if (channel->offset == -1) {
+    errno = 0; // clear
+    channel->offset = 0;
+  }
   caml_leave_blocking_section();
   channel->curr = channel->max = channel->buff;
   channel->end = channel->buff + IO_BUFFER_SIZE;
