--- asmcomp/arm64/emit.mlp
+++ asmcomp/arm64/emit.mlp
@@ -30,10 +30,10 @@ open Emitenv
 
 (* Names for special regs *)
 
-let reg_domain_state_ptr = phys_reg 25 (* x28 *)
-let reg_trap_ptr = phys_reg 23 (* x26 *)
-let reg_alloc_ptr = phys_reg 24 (* x27 *)
-let reg_tmp1 = phys_reg 26 (* x16 *)
+let reg_domain_state_ptr = phys_reg 24 (* x25 *)
+let reg_trap_ptr = phys_reg 22 (* x26 *)
+let reg_alloc_ptr = phys_reg 23 (* x27 *)
+let reg_tmp1 = phys_reg 25 (* x16 *)
 let reg_x8 = phys_reg 8 (* x8 *)
 
 (* Output a label *)
